app/main.py

```diff
@@ -1,2 +1,242 @@
+class Node:
+    key: Hashable
+    k_hash: int
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        # added mainly to keep track of input order, iteration
+        self._keys = []
+        self._values = []
+
+        self._initial_capacity = 8
+        self._load_factor = 2 / 3
+
+        self._capacity = self._initial_capacity
+        self._hash_table = [[] for _ in range(self._capacity)]
+
+    # didn't add formating for different types due to time complexity
+    def __str__(self) -> str:
+        pairs = []
+        for key, value in zip(self._keys, self._values):
+            pairs.append(f"{key}: {value}")
+        result = "{" + ", ".join(pairs) + "}"
+        return result
+
+    def __getitem__(self, key: Hashable) -> Any:
+        return self._data_handler(key)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+
+        if self.get_load():
+            self._data_handler(key, value)
+        else:
+            self.resize()
+            self.__setitem__(key, value)
+
+    def __len__(self) -> int:
+        return sum(1 for item in self._hash_table if item)
+
+    def __delitem__(self, key: Hashable) -> None:
+        if self.__getitem__(key):
+            self._data_handler(key, delete=True)
+
+    def keys(self) -> str:
+        class DictKeys:
+            def __init__(self, keys: list = None) -> None:
+                self._keys = keys if keys else []
+
+            def __iter__(self) -> Iterable:
+                return iter(self._keys)
+
+            def __repr__(self) -> str:
+                return f"dict_keys({self._keys})"
+
+        return DictKeys(self._keys)
+
+    def values(self) -> str:
+        class DictValues:
+            def __init__(self, values: list = None) -> None:
+                self._values = values if values else []
+
+            def __iter__(self) -> Iterable:
+                return iter(self._values)
+
+            def __repr__(self) -> str:
+                return f"dict_values({self._values})"
+
+        return DictValues(self._values)
+
+    def clear(self) -> None:
+        self._hash_table = [[] for _ in range(self._capacity)]
+        self._keys = []
+        self._values = []
+
+    def get(self, key: Hashable) -> Any:
+        try:
+            return self.__getitem__(key)
+        except KeyError:
+            return None
+
+    def pop(self, key: Hashable) -> Any:
+        value = self.__getitem__(key)
+        self.__delitem__(key)
+        return value
+
+    def __iter__(self) -> Dictionary:
+        self.current_key = 0
+        return self
+
+    def __next__(self) -> Hashable:
+        while self.current_key < len(self._keys):
+            key = self._keys[self.current_key]
+            self.current_key += 1
+            return key
+        raise StopIteration
+
+    def update(self, other: Dictionary) -> None:
+        for item in other:
+            self.__setitem__(item, other[item])
+
+    def _data_handler(
```

Comment:
Whatever is below this line is an example of so called spaghetti code :spaghetti:  You need to rethink the entire concept. 
Here's the algorytm you should follow with your set item method for instance:
1. Get hash value of key
2. Check if table needs to be resized and resize if necessary
3. Create a node with all necessary data
4. Find correct index for table cell, write data, increase counter of occupied table cells.
5. Code will be  pretty much the same for `__getitem__` method.
6. It should take you up to 30  lines of code to implement this logic including methods to resize table and traverse table to find right index.